tensorflow/contrib/avro/python/parse_avro_record.py

```diff
@@ -0,0 +1,476 @@
+                if dense_types[i] == dtypes.string:
+                    default_value = ""
+                elif dense_types[i] == dtypes.bool:
+                    default_value = False
+                else:  # Should be numeric type
+                    default_value = 0
+                default_value = ops.convert_to_tensor(default_value, dtype=dense_types[i])
+                # ************* END difference: This part is different from the originally copied code *****************
+            else:
+                # Reshape to a scalar to ensure user gets an error if they
+                # provide a tensor that's not intended to be a padding value
+                # (0 or 2+ elements).
+                key_name = "padding_" + re.sub("[^A-Za-z0-9_.\\-/]", "_", key)
+                default_value = ops.convert_to_tensor(
+                    default_value, dtype=dense_types[i], name=key_name)
+                default_value = array_ops.reshape(default_value, [])
+        else:
+            # This part is used by the FixedLenFeature
+            if default_value is None:
+                default_value = constant_op.constant([], dtype=dense_types[i])
+            elif not isinstance(default_value, ops.Tensor):
+                key_name = "key_" + re.sub("[^A-Za-z0-9_.\\-/]", "_", key)
+                default_value = ops.convert_to_tensor(
+                    default_value, dtype=dense_types[i], name=key_name)
+                default_value = array_ops.reshape(default_value, dense_shape)
+
+        dense_defaults_vec.append(default_value)
+
+    # Finally, convert dense_shapes to TensorShapeProto
+    dense_shapes = [shape.as_proto() for shape in dense_shapes]
+
+    # ******************** START difference: This part is different from the originally copied code ********************
+    outputs = _parse_avro_record(serialized=serialized,
+                                 sparse_keys=sparse_keys,
+                                 sparse_types=sparse_types,
+                                 dense_defaults=dense_defaults_vec,
+                                 dense_keys=dense_keys,
+                                 dense_shapes=dense_shapes,
+                                 schema=schema)
+    # ********************** END difference: This part is different from the originally copied code ********************
+
+    (sparse_indices, sparse_values, sparse_shapes, dense_values) = outputs
+
+    sparse_tensors = [
+        sparse_tensor.SparseTensor(ix, val, shape) for (ix, val, shape)
+        in zip(sparse_indices, sparse_values, sparse_shapes)]
+
+    return _construct_sparse_tensors_for_sparse_features(
+        features, dict(zip(sparse_keys + dense_keys, sparse_tensors + dense_values)))
+
+
+def _create_parse_avro_record_op_proto():
+    """
+    Creates the operator definition for the parse avro record function.
+
+    Side note: Once we integrate this into the TensorFlow source code this would be automatically generated.
```

Comment:
Can this be addressed? I'm guessing you need to create a tf_gen_op_wrapper_py BUILD rule?

Reply:
Addressed